### ISSUE TITLE
why dmi loss need to add '0.001' to det(mat)+0.001

### ISSUE THREAD

**Issue by yyht** [Newbeeer/L_DMI#9]
hi, since it needs torch.log to det(mat), usually we add 1e-10 to achieve numerical stability, why in this loss, we need to add 1e-3?

**Comment by dbp1994** [comment]
Numerical Stability to avoid degenerate case of det(mat)=0 thereby resulting in a log(0) scenario.

**Comment by Newbeeer** [comment]
Very sorry for the huge delay. Yes, @dbp1994 gets the point. We add the small pertubation to avoid the det(mat)=0 case.
